qhana_plugin_registry/api/models/generators/type_map.py

```diff
@@ -48,49 +48,84 @@ class ResourceMetadata:
     RootDataRaw: ResourceMetadata(
         rel_type=c.ROOT_REL_TYPE,
         extra_link_rels=c.ROOT_EXTRA_LINK_RELATIONS,
-        endpoint=c.ROOT_RESOURCE,
+        endpoint=None,
         schema=RootSchema,
         schema_id=RootSchema.schema_name(),
         collection_endpoint=None,
     ),
     Env: ResourceMetadata(
         rel_type=c.ENV_REL_TYPE,
         extra_link_rels=c.ENV_EXTRA_LINK_RELATIONS,
-        endpoint=c.ENV_RESOURCE,
+        endpoint=None,
         schema=EnvSchema,
         schema_id=EnvSchema.schema_name(),
-        collection_endpoint=c.ENV_COLLECTION_RESOURCE,
+        collection_endpoint=None,
     ),
     Service: ResourceMetadata(
         rel_type=c.SERVICE_REL_TYPE,
         extra_link_rels=c.SERVICE_EXTRA_LINK_RELATIONS,
-        endpoint=c.SERVICE_RESOURCE,
+        endpoint=None,
         schema=ServiceSchema,
         schema_id=ServiceSchema.schema_name(),
-        collection_endpoint=c.SERVICE_PAGE_RESOURCE,
+        collection_endpoint=None,
     ),
     WorkspaceTemplate: ResourceMetadata(
         rel_type=c.TEMPLATE_REL_TYPE,
         extra_link_rels=c.TEMPLATE_EXTRA_LINK_RELATIONS,
-        endpoint=c.TEMPLATE_RESOURCE,
+        endpoint=None,
         schema=TemplateSchema,
         schema_id=TemplateSchema.schema_name(),
-        collection_endpoint=c.TEMPLATE_PAGE_RESOURCE,
+        collection_endpoint=None,
     ),
     Seed: ResourceMetadata(
         rel_type=c.SEED_REL_TYPE,
         extra_link_rels=c.SEED_EXTRA_LINK_RELATIONS,
-        endpoint=c.SEED_RESOURCE,
+        endpoint=None,
         schema=SeedSchema,
         schema_id=SeedSchema.schema_name(),
-        collection_endpoint=c.SEED_PAGE_RESOURCE,
+        collection_endpoint=None,
     ),
     RAMP: ResourceMetadata(
         rel_type=c.PLUGIN_REL_TYPE,
         extra_link_rels=c.PLUGIN_EXTRA_LINK_RELATIONS,
-        endpoint=c.PLUGIN_RESOURCE,
+        endpoint=None,
         schema=PluginSchema,
         schema_id=PluginSchema.schema_name(),
-        collection_endpoint=c.PLUGIN_PAGE_RESOURCE,
+        collection_endpoint=None,
     ),
 }
+
+
+def populate_endpoints():
```

Comment:
Just fill the whole map inside the function. The map creates many issues of circular importing so it makes sense to do it late. And if everything is inside the function it becomes more readable again.